config/main.py

```diff
@@ -288,13 +288,10 @@ def _abort_if_false(ctx, param, value):
 
 def _stop_services():
     services = [
-        'dhcp_relay',
         'swss',
-        'snmp',
         'lldp',
         'pmon',
         'bgp',
```

Comment:
@jleveque can we also add swss.service dependency for bgp and lldp and eventually have similar behaviour as for teamd? The idea here is to have network and platform parts separated and fully managed by systemd. What do you think?

Reply:
@nazariig: Good suggestions. I'm all in favor of simplifying the service management. I opened a PR to add the same dependency for BGP here (https://github.com/Azure/sonic-buildimage/pull/3078), but as you can see from the review comments, it appears there are corner cases, so we will need to think about a more intricate approach.

Regarding LLDP, it's not as imperative that we restart it along with SwSS. Now that we have lldpmgrd, anytime SwSS restarts it should get notified of port state changes and configure LLDP appropriately. It's not as critical as dhcp_relay or radv which have no way of being updated to port changes, but the dependency wouldn't hurt and would help simplify the service management, so I'd like to get @lguohan's opinion on this.

Reply:
@jleveque If swss service restarts in warm mode does it restart teamd?

Reply:
As long as there is a call to `systemctl [re]start swss`, teamd and the other services which are configured with the same dependency on swss will also be [re]started.

Reply:
@jleveque I believe this breaks swss service level warm restart because teamd will recreate portchannels. Is it going to be fixed?

Reply:
@yxieca / @pavel-shirshov to comment on teamd warm restart.

Reply:
@jleveque teamd is not a dependent of swss from what I see in current code. I think you need to put teamd back.

Reply:
Discussed offline. teamd is still dependent on swss. We will need a more elegant dependency solution in sonic-buildimage to handle the warm-restart case. But this PR should still be valid.